pandas/io/packers.py

```diff
@@ -356,7 +356,7 @@ def encode(obj):
             return {'typ': 'block_manager',
                     'klass': obj.__class__.__name__,
                     'axes': data.axes,
-                    'blocks': [{'items': data.items.take(b.mgr_locs),
+                    'blocks': [{'items': b.mgr_locs.as_array,
```

Comment:
instead of changing this. I think better to add a new property ``locs`` (which is the ``.as_array``, e.g. the locations)